be/src/formats/orc/orc_chunk_reader.cpp

```diff
@@ -408,6 +408,12 @@ static void _try_implicit_cast(TypeDescriptor* from, const TypeDescriptor& to) {
         } else {
             from->type = LogicalType::TYPE_DECIMAL32;
         }
+    } else if (_broker_load_mode && !_strict_mode && t1 == LogicalType::TYPE_VARCHAR &&
+               t2 == LogicalType::TYPE_VARCHAR) {
+        // For broker load, the orc field length is larger than the maximum length of the starrocks field 
+        // will cause load failure in non-strict mode. Here we keep the maximum length of the orc field 
+        // the same as the maximum length of the starrocks field.
+        from->len = to.len;
```

Comment:
According to the definition of strict mode, it seems that the result should be a NULL.
@jaogoy PTAL.
https://docs.starrocks.io/en-us/2.3/loading/load_concept/strict_mode#:~:text=Strict%20mode%20works%20as%20follows%3A%201%20If%20strict,StarRocks%20loads%20qualified%20rows%20together%20with%20error%20rows.

Reply:
@rickif For non-strict mode， data can be loaded successfully, but overlimited columns should be null.